File: CODE/SimpleReticulate.py
# Copyright 2025 by Steven J. Keyes. All rights reserved.
# FILE: Example_15_SimpleReticulate.py
# DATE 2025-10-16
# DESCRIPTION: 

import pandas as pd
from nba_api.stats.static import players
from nba_api.stats.endpoints import playercareerstats
import time

def get_player_id(player_name):
    """
    Retrieves the player ID from the NBA API.
    Returns the player ID if found, otherwise None.
    """
    nba_players = players.get_players()

    player_id = []
    for player in nba_players:
        if player['full_name'] == player_name:
            player_id.append(player)

    
    if player_id:
        return player_id[0]['id']
    return None

id = get_player_id("LeBron James")

print(id)

File: CODE/SimpleReticulate.R
# Copyright 2025 by Steven J. Keyes. All rights reserved.
# FILE: SimpleReticulate.R
# DATE 2025-10-15
# DESCRIPTION: 
# This R program demonstrates a sophisticated use of the reticulate 
# library to integrate Python capabilities directly into an R 
# environment. It specifically sets up a workspace to interact with 
# NBA data using Python's nba_api.


library(reticulate)

py_install(c("nba_api", "pandas"))

py_config()

pandas_module <- reticulate::import("pandas")
nba_api       <- reticulate::import("nba_api")

# Source the Python script
reticulate::source_python("SimpleReticulate.py")

lebron_id <- get_player_id("LeBron James")

print(lebron_id)
